doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+
+Every `cv::gapi::ie::Params<>` object is related to a network specified by its
+template argument. We should pass there the network type we have described in
+the previous step.
+These objects are able to contain the paths to structure and weights files and
+a desirable backend which can be one of backends supported by InferenceEngine
+("CPU", "GPU", etc.).
+
+### 3) Network inference inside a graph {#gapi_fb_i_3}
+
+We are able to use the network in our pipline now! To do that, we should call
+`cv::gapi::infer<>()` setting the network type as a template argument to
+specify the network we want to infer:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp net_usg
+
+## Videosource processing via Streaming-API {#gapi_fb_streaming}
+
+There are no interest in using the most of computer vision pipelines to process
+just a single picture -- a big group of them are created to be launched with a
+videosource as an input. It is therefore essential for the G-API as an
+interface for pipelines implementation to provide a possibility of a stream
+processing without any external instruments to be applied by user.
+So, that's why Streaming-API is needed.
+
+The usage of this G-API feature makes codewriting easier, or, at least, more
+independent. As usual in this tutorial, let's deal with it by several steps:
+
+### 1) Pipeline graph definition {#gapi_fb_s_1}
+
+To begin with, a `cv::GComputation` object (which contains all the
+pipeline) used to be defined statically by graph's input and output described in
+a code (almost all the graph's body is skipped in this code snippet except the
+input and output nodes):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_4
+
+To apply Streaming-API we should use a different version of a
+pipeline expression  with a lambda-based constructor to keep all temporary data
+objects in a dedicated scope:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_4
+
+### 2) Frames processing {#gapi_fb_s_2}
+
+The next step used to be a videostream definition and a source setup:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp header
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp in_out_1
+
+and then a compiled graph used to be applied to every single frame.
```

Comment:
`used to be` is incorrect here

Reply:
We should talk about that `used to be` thing, I can suppose I'm wrong. But at the moment I'm confident that it is the case to use this construction.